main.rb

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env ruby
+require_relative 'modules/appfunctions'
+
+class App
+  include AppFunctions
+  attr_reader :book_list, :people_list, :rental_list
+
+  def initialize
+    @book_list = []
+    @people_list = []
+    @rental_list = []
+    process_user_choice
+  end
+
+  def start
+    number = ''
+    while number == ''
+      puts "\nWelcome to the School Library App"
+      puts "\n"
+      puts 'Please choose an option by entering a number:'
+      puts '1 - List all books'
+      puts '2 - List all people'
+      puts '3 - Create a person'
+      puts '4 - Create a book'
+      puts '5 - Create a rental'
+      puts '6 - List all rentals for a given person id'
+      puts '7 - Exit'
+      number = gets.chomp.to_i
+      number = '' unless number.is_a?(Numeric) && number.positive? && number <= 7
+      puts 'Invalid selection' unless number != ''
+    end
+    number
+  end
+
+  def process_user_choice
+    selection = start
+    case selection
+    when 1
+      display_books
+    when 2
+      display_persons
+    when 3
+      create_person
+    when 4
+      create_book
+    when 5
+      add_rental
+    when 6
+      display_rentals
+    else
+      puts 'Closing library.....'
+      return
+    end
+    process_user_choice
+  end
+
+  def person_created
+    puts "Person created successfully\n"
+  end
+
+  def create_person
+    print 'Do you want to create a Student (1) or a Teacher (2)? '
+    print '[input the number]: '
+    number = gets.chomp.to_i
+
+    if number.is_a?(Numeric) && number.positive? && number <= 2
+      case number
+      when 1
+        @people_list.push(create_student)
+        person_created
+      when 2
+        @people_list.push(create_teacher)
+        person_created
+      end
+    else
+      puts 'Invalid selection'
+      create_person
+    end
+  end
+
+  def create_book
+    @book_list.push(create_new_book)
+    puts "Book created successfully\n"
+  end
+
+  def display_rentals
+    person_id = validate_id
+    person = get_person_with_id(person_id)
+
+    puts 'RENTALS:'
+    if person != ''
+      person.rentals.each_with_index do |rental, _i|
+        puts "Date: #{rental.date}, Book: '#{rental.book.title}', Author: '#{rental.book.author}'"
+      end
+    end
+    puts "\n"
+  end
+
+  def display_books
+    puts 'BOOKS:'
+    @book_list.each_with_index do |book, i|
+      puts "(#{i}) Title: #{book.title}, Author: #{book.author}"
+    end
```

Comment:
- [ ] Everything is working fine but When I tried to create a rental **second** time, I got the error of undefined method 'title'. which I have attached below.

![Screenshot from 2021-12-17 18-38-15](https://user-images.githubusercontent.com/22434660/146549620-59944c85-3161-4508-8dae-12e4d33060ba.png)

Reply:
Thanks a lot for spotting the error I was saving the rentals into the wrong List.

It should work perfectly now.
🙇